rfcs/select-component.md

```diff
@@ -1,92 +1,43 @@
-  * `Option: ?React.ComponentType` component to use for options in dropdown list
-  * `Root: ?React.ComponentType` component to use for most top of the select component
-  * `Input: ?React.ComponentType` component for Input showing current selected value(s). See `Input` Control of this framework for reference to override it's functionality.
-  * `SearchIcon: ?React.ComponentType` component for all icons appearing in Select component. It's provided `$type: ICON.loop | ICON.clearTag | ICON.clearAll | ICON.selected` to setup corresponding icon of Select component
-  * `Tag: ?React.ComponentType` component for selected options Tags shown in Input for multiple mode selection
-* `onChange: func(e: SyntheticInputEvent, params: Object)`:
-  handler for events on trigger element when option are changing selection or text of search input (in Search mode has changed). `params` has `type` indicating which action is performed, `id` and `label` of selected\unselected option and `selectedOptions` array of all of selected, has new `textValue` set in input (for `keyUp`)
-* `onMouseEnter: func`:
-  handler for events on trigger element
-* `onMouseLeave: func`:
-  handler for events on trigger element
-* `onFocus: func`:
-  handler for events on trigger element
-* `onBlur: func`:
-  handler for events on trigger element
-
-## `StatefulSelectContainer` API
-
-* `initialState: {}`
-  Initial state of an uncontrolled popover component.
-  * `selectedOptions` - an initial set of selected options. They are prepended to all Options array if not found there.
-* `stateReducer: (type: text, nextState: {}, currentState: {}, e: any, params: Object) => nextState`
-  A state change handler.
-  * `type` - state change type
-  * `nextState` - a new state changes that will be set
-  * `currentState` - current full state of the component
-  * `params` may contain `id` and `label` of selected option and `selectedOptions` array of all of selected, as well as new `textValue` set in input.
-* `options: func` should return `Select` instance with standard or customized inner elements.
-* `onChange: func(e: SyntheticInputEvent, params: Object)`:
-  handler for events on trigger element when option are changing selection or text of search input (in Search mode has changed). `params` has `type` indicating which action is performed, `id` and `label` of selected\unselected option and `selectedOptions` array of all of selected, has new `textValue` set in input (for `keyUp`)
-* `onMouseEnter: func`:
-  handler for events on trigger element
-* `onMouseLeave: func`:
-  handler for events on trigger element
-* `onFocus: func`:
-  handler for events on trigger element
-* `onBlur: func`:
-  handler for events on trigger element
+export default () => {
+  const options = [
+    {
+      id: '1',
+      label: {
+        text: 'First',
+        imgSrc: '1.jpg',
+      },
+    },
+    {
+      id: '2',
+      disabled: true,
+      label: {
+        text: 'Second',
+        imgSrc: '2.jpg',
+      },
```

Comment:
We should use string labels here – i'm not sure the object format works without specifying a `getOptionLabel` callback.